api/envoy/api/v2/route/route.proto

```diff
@@ -615,7 +615,22 @@ message RouteAction {
     // 0.01% of requests to be mirrored. If the runtime key is specified in the
     // configuration but not present in runtime, 0 is the default and thus 0% of
     // requests will be mirrored.
-    string runtime_key = 2;
+    string runtime_key = 2 [deprecated = true];
+
+    // If not specified, all requests to the target cluster will be mirrored. If specified,
+    // requests must also fall under the percentage of matches indicated by this field.
+    // For some fraction N/D, a random number in the range [0,D) is selected. If the
+    // number is <= the value of the numberator N, or if the key is not present, the default
+    // value, the request will be mirrored.
+    //
+    // .. note::
+    //
+    //    Parsing this field is implemented such that the runtime key's data may be represented
+    //    as a FractionalPercent proto represented as JSON/YAML and may also be represented as an
+    //    integer with the assumption that the value is an integral percentage out of 100. For
+    //    instance, a runtime key lookup returning the value "42" would parse as a FractionalPercent
+    //    whose numerator is 42 and denominator is HUNDRED. This preserves legacy semantics.
```

Comment:
I think this comment is incorrect. The existing runtime_key is in the range 0-10000, so a runtime_fraction.runtime_key lookup returning "42" will shadow 42% of the time vs 42/10000 = 0.42% for runtime_key.

I think we'll need to replace this section with an explanation of how to leg over, which I think requires setting up a new key that uses the fractional proto format for its value.